frontend/pulse3d/src/pages/production-console.js

```diff
@@ -1,22 +1,725 @@
+  padding-bottom: 30px;
 `;
 
-const SectionHeader = styled.div`
+const TableHeader = styled.div`
   padding-bottom: 10px;
   padding-left: 10px;
 `;
 
-export default function ProductionConsole() {
+const serNumScopes = ["mantarray:serial_number:edit"];
+const fwScopes = ["mantarray:firmware:edit", "mantarray:firmware:info"];
+
+const fwTypes = ["Main", "Channel"];
+
+function FirmwareSection({ accountScope }) {
+  const canViewFwTables = accountScope.includes("mantarray:firmware:info");
+  const canEditFw = accountScope.includes("mantarray:firmware:edit");
+
+  const [fwInfo, setFwInfo] = useState({ main: [], channel: [] });
+
+  const getFwInfo = async () => {
+    const getFwInfoRes = await fetch(`${process.env.NEXT_PUBLIC_MANTARRAY_URL}/firmware/info`);
+    const getFwInfoResJson = await getFwInfoRes.json();
+    setFwInfo({ main: getFwInfoResJson.main_fw_info, channel: getFwInfoResJson.channel_fw_info });
+  };
+
+  // when component first loads, get FW info
+  useEffect(() => {
+    getFwInfo();
+  }, []);
+
   return (
-    <Container>
-      <SectionHeader>Mantarray Serial Numbers</SectionHeader>
+    <>
+      {canEditFw && <FirmwareUpload fwInfo={fwInfo} refreshTables={getFwInfo} />}
+      {canViewFwTables && <FirmwareTables fwInfo={fwInfo} canEdit={canEditFw} getFwInfo={getFwInfo} />}
+    </>
+  );
+}
+
+const UPLOAD_STATES = {
+  IDLE: "IDLE",
+  IN_PROGRESS: "IN_PROGRESS",
+  SUCCESS: "SUCCESS",
+  FAILED: "FAILED",
+};
+
+const dropZoneText = "CLICK HERE or DROP";
+
+function FirmwareUpload({ fwInfo, refreshTables }) {
+  const [file, setFile] = useState();
+  const [resetDragDrop, setResetDragDrop] = useState(false);
+  const [resetInputWidgets, setResetInputWidgets] = useState(false);
+  const [uploadOptions, setUploadOptions] = useState({});
+  const [uploadOptionErrors, setUploadOptionErrors] = useState({});
+  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+  const [uploadState, setUploadState] = useState(UPLOAD_STATES.IDLE);
+
+  const mainFwVersions = fwInfo.main.map((row) => row.version).sort(semverSortFn);
+  const inProgress = uploadState === UPLOAD_STATES.IN_PROGRESS;
+
+  useEffect(() => {
+    const canUpload =
+      getFileName(file) &&
+      fwTypes.includes(uploadOptions.fwType) &&
+      uploadOptions.version &&
+      noErrors() &&
+      checkFwTypeSpecificOptions();
+
+    setIsButtonDisabled(!canUpload || inProgress);
+  }, [file, uploadOptions, uploadOptionErrors, inProgress]);
+
+  useEffect(() => {
+    if (resetInputWidgets) {
+      setResetInputWidgets(false);
+    }
+  }, [resetInputWidgets]);
+
+  const resetInputs = () => {
+    setFile(null);
+    setResetInputWidgets(true);
+    setUploadOptions({});
+    setUploadOptionErrors({});
+  };
+
+  const getFileName = (file) => {
+    return file ? file.name : null;
+  };
+
+  const noErrors = () => {
+    return Object.values(uploadOptionErrors).every((val) => val.length === 0);
+  };
+  fwTypes.includes(uploadOptions.fwType);
```

Comment:
is this meant to be here?

Reply:
Not sure, looks like it's not doing anything though